File: algorithms/python/SelectionSort.py
def selectionSort(array, size):
   
    for step in range(size):
        min_idx = step

        for i in range(step + 1, size):
            if array[i] < array[min_idx]:
                min_idx = i
         
        # putting min at the correct position
        (array[step], array[min_idx]) = (array[min_idx], array[step])


data = [-2, 45, 0, 11, -5]
size = len(data)
selectionSort(data, size)
print('Sorted Array in Ascending Order:')
print(data)

File: algorithms/javascript/selection_sort.js
function selectionSort(array) {
    for (let i = 0; i < array.length; i++) {
      let minIndex = i;
  
      for (let j = i + 1; j < array.length; j++) {
        if (array[j] < array[minIndex]) {
          minIndex = j;
        }
      }
  
      [array[i], array[minIndex]] = [array[minIndex], array[i]];
    }
  
    return array;
  }